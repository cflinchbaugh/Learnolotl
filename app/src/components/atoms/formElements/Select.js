import React from 'react';
import styled from 'styled-components';

const StyleWrapper = styled.div`
    
`

function Select (props) {
    let options = props.options.map( (option, idx) => {
            return (
                <option key={idx} value={option.id}>
                    {option.value}
                </option>
            );
        });
    
    return (
        <StyleWrapper>
            <div>{props.label}</div>
            <select id={props.id} onChange={props.handleChange}>
                {options}
            </select>
        </StyleWrapper>
    )
}

export default Select