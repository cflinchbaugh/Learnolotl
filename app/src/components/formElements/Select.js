import React from 'react';
import styled from 'styled-components';

const StyleWrapper = styled.div`
    
`

function Select (props) {
    let options = props.options.map( (option, idx) => {
            let optionData = {
                key: idx,
                value: option.id
            }

            return (
                <option {...optionData}>
                    {option.value}
                </option>
            );
        });
    
    return (
        <StyleWrapper>
            <div>{props.label}</div>
            <select id={props.id} onChange={props.handleChange} value={props.selected}>
                {options}
            </select>
        </StyleWrapper>
    )
}

export default Select