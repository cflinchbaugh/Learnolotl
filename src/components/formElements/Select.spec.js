import React from 'react';
import {shallow} from 'enzyme';
import Select from './Select';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Select', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const selectData = {
            options: []
        }

        const tree = renderer.render(
            <Select {...selectData}/>
        );

        expect(tree).toMatchSnapshot();
    });

    it('renders populated data and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const selectData = {
            label: 'DC',
            id: 'DC-Selector',
            handleChange: jest.fn(),
            options: [
                {
                    id: 'Wayne',
                    value: 'Batman'
                }, {
                    id: 'Grayson',
                    value: 'Robin'
                }
            ]
        }

        const tree = renderer.render(
            <Select {...selectData}/>
        );

        expect(tree).toMatchSnapshot();
    });

});