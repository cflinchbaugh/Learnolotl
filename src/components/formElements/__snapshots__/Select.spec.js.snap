// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders and matches snapshot 1`] = `
<styled.div>
  <div>
    :
    <select
      id={undefined}
      onChange={undefined}
      value={undefined}
    >
      
    </select>
  </div>
</styled.div>
`;

exports[`Select renders populated data and matches snapshot 1`] = `
<styled.div>
  <div>
    DC
    :
    <select
      id="DC-Selector"
      onChange={[Function]}
      value={undefined}
    >
      <option
        value="Wayne"
      >
        Batman
      </option>
      <option
        value="Grayson"
      >
        Robin
      </option>
    </select>
  </div>
</styled.div>
`;
